action/action_deserializer.go

```diff
@@ -9,12 +9,25 @@ package action
 import "github.com/iotexproject/iotex-proto/golang/iotextypes"
 
 // Deserializer de-serializes an action
+//
+// It's a wrapper to set certain parameters in order to correctly de-serialize an action
+// Currently the parameter is EVM network ID for tx in web3 format, it is called like
+//
+// act, err := (&Deserializer{}).SetEvmNetworkID(id).ActionToSealedEnvelope(pbAction)
+//
 type Deserializer struct {
+	evmNetworkID uint32
+}
+
+// SetEvmNetworkID sets the evm network ID for web3 actions
+func (ad *Deserializer) SetEvmNetworkID(id uint32) *Deserializer {
+	ad.evmNetworkID = id
+	return ad
 }
 
 // ActionToSealedEnvelope converts protobuf to SealedEnvelope
 func (ad *Deserializer) ActionToSealedEnvelope(pbAct *iotextypes.Action) (SealedEnvelope, error) {
 	var selp SealedEnvelope
-	err := selp.LoadProto(pbAct)
+	err := selp.LoadProto(pbAct, ad.evmNetworkID)
```

Comment:
Since Deserializer has to be used, `selp.LoadProto` can be changed to private func?

Reply:
will file a new PR to do this